samza-core/src/main/java/org/apache/samza/clustermanager/ContainerProcessManager.java

```diff
@@ -97,6 +100,8 @@
 
   // The ContainerManager manages control actions for both active & standby containers
   private final ContainerManager containerManager;
+  private ContainerPlacementHandler containerPlacementHandler;
+  private Thread containerPlacementHandlerThread;
```

Comment:
Please use ExecutorService executorService = Executors.newSingleThreadExecutor(

Reply:
Maybe we can use the 
```afterExecute(Runnable r, Throwable t)``` method of ExecutorService?

Reply:
nope I have to use the semaphore for propagating the exception from CPH thread to main thread